src/main/kotlin/cdsre/utils/Endianness.kt

```diff
@@ -62,19 +111,47 @@ class Endianness {
          */
         public fun fromUInt(uint: UInt, endianness: Endian): IntArray{
             val int = uint.toInt()
+            return toIntArray(int, 4, endianness)
+        }
+        
+        /**
+         * @param ulong unsigned long to unpack
+         * @param endianness which endian to unpack bytes by
+         * @return an array containing the bytes ordered by endianness
+         */
+        public fun fromULong(ulong: ULong, endianness: Endian): IntArray {
+            val long = ulong.toLong()
+            return longToIntArray(long, endianness)
+        }
+        
+        /**
+         * Unpack any value of size int or smaller into an array
+         * @param int the value to unpack into an array
+         * @param endianness the endianness by which to unpack
+         * @param bytes the number of bytes to unpack
+         * @return the array of the bytes ordered by endianness
+         */
+        private fun toIntArray(int: Int, bytes: Int, endianness: Endian): IntArray {
+            if(endianness == Endian.LITTLE) {
+                return IntArray(bytes){ i -> ((int shr (i*8)) and 0xFF).toInt()}
+            } else {
+                return IntArray(bytes){ i -> ((int shr ((bytes-1-i)*8)) and 0xFF).toInt()}
+            }
+        }
+        
+        /**
+         * Long version of {@link toIntArray}
+         * @param long the value to unpack into an array
+         * @param endianness the endianness by which to unpack
+         * @return the array of the bytes ordered by endianness
+         */
+        private fun longToIntArray(long: Long, endianness: Endian): IntArray {
             if(endianness == Endian.LITTLE) {
-                return intArrayOf(
-                        ((int shr 0) and 0xFF),
-                        ((int shr 8) and 0xFF),
-                        ((int shr 16) and 0xFF),
-                        ((int shr 24) and 0xFF))
+                return IntArray(8){ i -> ((long shr (    i*8)) and 0xFF).toInt()}
```

Comment:
Why the extra spacing? So it matches up with the next line, I'm guessing?

Reply:
Correct :)